src/RA_AchievementSet.cpp

```diff
@@ -480,119 +442,121 @@ BOOL AchievementSet::LoadFromFile(ra::GameID nGameID)
+        const auto& LeaderboardsData{ doc["Leaderboards"] };
+        for (const auto& lbData : LeaderboardsData.GetArray())
+        {
+            //"Leaderboards":[{"ID":"2","Mem":"STA:0xfe10=h0000_0xhf601=h0c_d0xhf601!=h0c_0xfff0=0_0xfffb=0::CAN:0xhfe13<d0xhfe13::SUB:0xf7cc!=0_d0xf7cc=0::VAL:0xhfe24*1_0xhfe25*60_0xhfe22*3600","Format":"TIME","Title":"Green Hill Act 1","Description":"Complete this act in the fastest time!"},
 
-                    lb.SetTitle(lbData["Title"].GetString());
-                    lb.SetDescription(lbData["Description"].GetString());
+            RA_Leaderboard lb{ lbData["ID"].GetUint() };
 
-                    auto nFormat = MemValue::ParseFormat(lbData["Format"].GetString());
-                    lb.ParseFromString(lbData["Mem"].GetString(), nFormat);
+            lb.SetTitle(lbData["Title"].GetString());
+            lb.SetDescription(lbData["Description"].GetString());
 
-                    g_LeaderboardManager.AddLeaderboard(lb);
-                }
-            }
-            else
-            {
-                fclose(pFile);
-                ASSERT(!"Could not parse file?!");
-                return FALSE;
-            }
+            const auto nFormat{ MemValue::ParseFormat(lbData["Format"].GetString()) };
+            lb.ParseFromString(lbData["Mem"].GetString(), nFormat);
 
-            fclose(pFile);
+            g_LeaderboardManager.AddLeaderboard(lb);
+        }
+
+        // calculate the total number of points for the core set, and pre-fetch badge images
+        auto nTotalPoints{0U};
+        for (size_t i = 0; i < g_pCoreAchievements->NumAchievements(); ++i)
+        {
+            const auto& ach{ g_pCoreAchievements->GetAchievement(i) };
+            ra::services::g_ImageRepository.FetchImage(ra::services::ImageType::Badge, ach.BadgeImageURI());
+            nTotalPoints += ach.Points();
+        }
 
-            // calculate the total number of points for the core set, and pre-fetch badge images
-            unsigned int nTotalPoints = 0;
-            for (size_t i = 0; i < g_pCoreAchievements->NumAchievements(); ++i)
+        if (RAUsers::LocalUser().IsLoggedIn())
+        {
+            //	Loaded OK: post a request for unlocks
+            PostArgs args;
+            args['u'] = RAUsers::LocalUser().Username();
+            args['t'] = RAUsers::LocalUser().Token();
+            args['g'] = std::to_string(nGameID);
+            args['h'] = g_bHardcoreModeActive ? "1" : "0";
+
+            RAWeb::CreateThreadedHTTPRequest(RequestUnlocks, args);
+            std::string sTitle;
             {
-                Achievement& ach = g_pCoreAchievements->GetAchievement(i);
-                ra::services::g_ImageRepository.FetchImage(ra::services::ImageType::Badge, ach.BadgeImageURI());
-                nTotalPoints += ach.Points();
+                std::ostringstream oss;
+                oss << "Loaded " << g_pCurrentGameData->GameTitle();
+                sTitle = oss.str();
```

Comment:
I just realized that `AchievementSet` already had a member function but this was static. The main reason for doing it this way is limit the number of reallocs. Plus the string from the stream is moved (O(1)) instead of copied (O(N)). 

It's kinda similar in C#, with the using scope but I'm sure the people here already know.